gb-chris/scripts/article.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+function Article (rawDataObj) {
+  this.author = rawDataObj.author;
+  this.authorUrl = rawDataObj.authorUrl;
+  this.title = rawDataObj.title;
+  this.category = rawDataObj.category;
+  this.body = rawDataObj.body;
+  this.publishedOn = rawDataObj.publishedOn;
+}
+
+// REVIEW: Instead of a global `articles = []` array, let's attach this list of all articles directly to the constructor function. Note: it is NOT on the prototype. In JavaScript, functions are themselves objects, which means we can add properties/values to them at any time. In this case, the array relates to ALL of the Article objects, so it does not belong on the prototype, as that would only be relevant to a single instantiated Article.
+Article.all = [];
+
+// COMMENT: Why isn't this method written as an arrow function?
+// The method uses a contextualized this that requires to be scoped towards the Article instances, making it an arrow function would scope the this to be on window at this level.
+Article.prototype.toHtml = function() {
+  let template = Handlebars.compile($('#article-template').text());
+
+  this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
+
+  // COMMENT: What is going on in the line below? What do the question mark and colon represent? How have we seen this same logic represented previously?
+  // Not sure? Check the docs!
+  // This is a turnary operator that functions as a conditional. The statement is saying IF this(Article instance).publishedOn is true evaluate to the template literal, otherwise the published status is draft.
+  this.publishStatus = this.publishedOn ? `published ${this.daysAgo} days ago` : '(draft)';
+  this.body = marked(this.body);
+
+  return template(this);
+};
+
+// REVIEW: There are some other functions that also relate to all articles across the board, rather than just single instances. Object-oriented programming would call these "class-level" functions, that are relevant to the entire "class" of objects that are Articles.
+
+// REVIEW: This function will take the rawData, how ever it is provided, and use it to instantiate all the articles. This code is moved from elsewhere, and encapsulated in a simply-named function for clarity.
+
+// COMMENT: Where is this function called? What does 'rawData' represent now? How is this different from previous labs?
+// The rawData here is now in local storage. Unlike prior labs where our raw data was a local .js file that we drew our articles from. It is now a local storage item that we draw out or load from a server.
```

Comment:
It's a local storage item for reloads of the page, but on the first load of the page, we're still using that separate .json file to pull it in! The difference is that, instead of including that json file in our `html` in a `script` tag, we're requesting the `json` data asynchronously using `$.ajax`.